### PR TITLE
Refactor AddIconsToReturnRector, AddArgumentToSymfonyCommandRector and AddCommandsToReturnRector to proper node factories

### PR DESCRIPTION
**Pull request by TomasVotruba**
Like controller cannot include another controller, the rule cannot use another rule for refactoring. Things can go wrong :)
The correct way would be factory or simple node traverser :+1: 

Discoverd during https://github.com/rectorphp/rector-src/pull/2239

### CODE REVIEW

File: src/Rector/v11/v5/RegisterIconToIconFileRector.php [hunk 1]
```diff
@@ -164,26 +163,24 @@ private function addNewIconToIconsFile(
...
+
         if (is_string($existingIcons)) {
             $stmts = $this->simplePhpParser->parseString($existingIcons);
+            $this->traverseNodesWithCallable($stmts, function (Node $node) use ($iconArrayItem) {
+                if (! $node instanceof Array_) {
+                    return null;
+                }
+
+                $node->items[] = $iconArrayItem;
+                return NodeTraverser::DONT_TRAVERSE_CHILDREN;
```
Comment (TomasVotruba) [inline src/Rector/v11/v5/RegisterIconToIconFileRector.php:176]: Add it just once, avoids infinity loop

**Comment by TomasVotruba** [comment]
Thanks for fast merge 👍

Could you verify it works as expected? 😆 I think the tests only verify existence of file, not exact contents.

**Comment by sabbelasichon** [comment]
The tests are also checking the contents:
https://github.com/sabbelasichon/typo3-rector/blob/main/tests/Rector/v9/v5/ExtbaseCommandControllerToSymfonyCommandRector/ExtbaseCommandControllerToSymfonyCommandRectorTest.php#L28

https://github.com/sabbelasichon/typo3-rector/blob/main/tests/Rector/v11/v5/RegisterIconToIconFileRector/RegisterIconToIconFileRectorTest.php#L29

Well done.

**Comment by TomasVotruba** [comment]
Ah, that's great. I was almost sleeping at 2 AM, so I was just surprised test were passing so easily :laughing: 


Thanks :+1:

**Pull request merged**